lib/irb/type_completion/types.rb

```diff
@@ -0,0 +1,448 @@
+# frozen_string_literal: true
+
+module IRB
+  module TypeCompletion
+    module Types
+      singleton_class.attr_reader :rbs_builder, :rbs_load_error
+
+      def self.preload_in_thread
+        @loader_thread ||= Thread.new do
```

Comment:
In "production" environment, this will only be called once. Do we need to memoize it?

Reply:
I think it has some benefits.

The original pull-request came to katakata_irb is an option of async loading because in large project, it takes more than seconds.
The default is async, but users can change to sync, always complete loading before IRB starts, by writing `IRB::TypeCompletion::Types.preload_in_thread.join` to irbrc.
I think it will help when we enable switching completors with irb commands like `irb(main):001> conf.completor = :type`.

I also think it's safer. (for multi-irb, executing `binding.irb` inside irb, `binding.irb; binding.irb` or other case)

Reply:
IMO `IRB::TypeCompletion::Types.preload_in_thread` should be considered private. If we want to let users control async/sync loading, we should encapsulate it under our config design.

> I think it will help when we enable switching completors with irb commands

I don't think it's very useful to switch completor mid-session other than testing/debugging IRB itself?

Reply:
Agree that it should be private.

Actually, in production running this ruby file, preload_in_thread is ran twice.
```ruby
# run with `ruby a.rb`
2.times do
  binding.irb
end
```
Typing `exit` fast with slow_loading environment, (many entry in rbs_collection.yaml and sig directory), preload_in_thread is ran twice before `@rbs_builder` is assigned.

To avoid preloading twice, we need to cache something outside of thread because thread creation can take a while. Cannot use `@rbs_builder` for this because assigning to it will delay. Other way is to see the return value of require and run preload_in_thread only if it is true.


In test, directly calling load_rbs_builder looks good. I'm thinking to change it to
```ruby
def self.preload_in_thread
  return if @preload_started
  @preload_started = true
  Thread.new do
    load_rbs_builder
  end
end
```